app/connectors/PayApiConnector.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.google.inject.Inject
+import config.ConfigDecorator
+import models.PaymentRequest
+import play.api.http.Status._
+import play.api.libs.json.{Json, Reads}
+import services.http.WsAllMethods
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final case class CreatePayment(journeyId: String, nextUrl: String)
+
+object CreatePayment {
+  implicit val format = Json.format[CreatePayment]
+}
+
+sealed trait CreatePaymentResponse
+final case class CreatePaymentSuccess(createPaymentResponse: CreatePayment) extends CreatePaymentResponse
+object CreatePaymentFailed extends CreatePaymentResponse
+
+class PayApiConnector @Inject()(http: WsAllMethods, configDecorator: ConfigDecorator) {
+
+  def createPayment(
+    request: PaymentRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[CreatePaymentResponse] = {
+    val postUrl = configDecorator.makeAPaymentUrl
+
+    http.POST[PaymentRequest, HttpResponse](postUrl, request) flatMap { response =>
+      response.status match {
+        case CREATED =>
+          val createPayment = responseTo[CreatePayment](response)
+          Future.successful(new CreatePaymentSuccess(createPayment))
+        case _ => Future.successful(CreatePaymentFailed)
```

Comment:
Why have you have you chosen to create your own success/failure datatype?
You have a few options available that will give greater usability.

1. Use the future failure channel with a bad request exception
This will allow consumers to not have to consider the failure case of this method as it will automatically short circuit any computations after a failure. The downside is you have to use a `BadRequestException` to have the global error handler interpret it correctly which means this method would need knowledge of how the application is structured as a whole.

2. Use `Either[Error, CreatePayment]` datatype
You get all the good stuff that comes with either like map, fold, flatMap, filter etc which will make the api of the class easier to work with. It also works well when there are multiple failure cases

3. Use `Option[CreatePayment]` datatype
Probably the one I would go with. This can represent the exact same use case that you have implemented here with the added benefit that you get all the good methods that come with option and everyone should know how to deal with them. Of course of there are multiple ways that a method can fail option cannot report on that